### PR TITLE
make AttrVal tokens() and atom() return or panic

### LINKED ISSUE #7479: Make AttrValue's tokens() and atom() return or panic
There is no reason an AttrValue should change variant once set, so we should refactor the API accordingly. This allows us to have better consistent panic messages in such cases.

### PR DESCRIPTION
**Pull request by psdh**
Fixes #7479

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7511)

### CODE REVIEW

**Comment by nox** [comment]
Please rename to as_atoms and as_tokens.

**Comment by nox** [comment]
@psdh This looks good, but #7452 is going to bit-rot it, and I introduced a uint() method there. You are going to need to rebase when the other PR is merged, could you rename uint() accordingly when it is time?

File: components/script/dom/domtokenlist.rs [hunk 1]
```diff
@@ -69,7 +69,7 @@ impl DOMTokenListMethods for DOMTokenList {
     fn Item(&self, index: u32) -> Option<DOMString> {
         self.attribute().and_then(|attr| {
             let attr = attr.r();
-            attr.value().tokens().and_then(|tokens| {
+            Some(attr.value().as_tokens()).and_then(|tokens| {
```
Comment (eefriedman) [inline components/script/dom/domtokenlist.rs]: This could be cleaned up.

**Comment by jdm** [comment]
@bors-servo: retry
* #7422

**Pull request merged**